File: backend/app/schemas/reservation.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.models.enums import ReservationStatus


class ReservationCreateRequest(BaseModel):
    product_id: int
    quantity: int = Field(gt=0)


class ReservationItemDetail(BaseModel):
    product_id: int
    title: str
    quantity: int
    price: int
    original_price: int

    model_config = {"from_attributes": True}


class ReservationDetail(BaseModel):
    id: int
    reservation_number: str
    status: ReservationStatus
    total_price: int
    pickup_time: datetime
    created_at: datetime
    items: list[ReservationItemDetail]

    model_config = {"from_attributes": True}

File: frontend/src/lib/api/reservation.ts
import { apiFetch } from "./client";

export type ReservationItem = {
  product_id: number;
  title: string;
  quantity: number;
  price: number;
};

export type Reservation = {
  id: number;
  reservation_number: string;
  status: string;
  total_price: number;
  pickup_time: string;
  created_at: string;
  items: ReservationItem[];
};

export function createReservation(
  userId: string,
  productId: number,
  quantity: number,
): Promise<Reservation> {
  return apiFetch<Reservation>("/reservations", {
    method: "POST",
    headers: { "X-User-Id": userId },
    body: JSON.stringify({ product_id: productId, quantity }),
  });
}

export function getReservation(userId: string, reservationId: number): Promise<Reservation> {
  return apiFetch<Reservation>(`/reservations/${reservationId}`, {
    headers: { "X-User-Id": userId },
  });
}
